db/db_properties_test.cc

```diff
@@ -1713,6 +1716,44 @@ TEST_F(DBPropertiesTest, BlockCacheProperties) {
   ASSERT_EQ(0, value);
 }
 
+TEST_F(DBPropertiesTest, TableCacheProperties) {
+  Options options;
+  uint64_t value, new_value;
+
+  options.env = CurrentOptions().env;
+
+  // Block cache properties are not available if block cache is not used.
+  BlockBasedTableOptions table_options;
+  options.table_factory.reset(NewBlockBasedTableFactory(table_options));
+  Reopen(options);
+
+  //
+  // test table_cache access is "live"
+  //  get default max_open_files
+  //
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kTableCacheCapacity, &value));
+  new_value = value / 2;
+
+  std::unordered_map<std::string, std::string> new_options;
+  new_options.insert(std::pair<std::string, std::string>(
+      "max_open_files", std::to_string(new_value)));
+  rocksdb::Status stat = db_->SetDBOptions(new_options);
+  ASSERT_TRUE(stat.ok());
+  ASSERT_TRUE(db_->GetIntProperty(DB::Properties::kTableCacheCapacity, &value));
+  // rocksdb does a -10 to number passed in
+  ASSERT_EQ(new_value - 10, value);
```

Comment:
I do not understand how the math works in this test.  You get the value, divide it by 2, subtract 10, and add 1.  By just reading the test this appears to be some sort of odd voodoo.

Can you add some comments somewhere to explain how/why this math works and what you are measuring?

Reply:
Will add comments.  In short, there are two different properties being tested "capacity" and "usage".